src/earn/saga.ts

```diff
@@ -338,77 +338,72 @@ export function* withdrawSubmitSaga(action: PayloadAction<WithdrawInfo>) {
-    const createWithdrawStandbyTxHandler = (
-      transactionHash: string,
-      feeCurrencyId?: string
-    ): BaseStandbyTransaction => {
-      return {
+    if (mode !== 'claim-rewards') {
+      const createWithdrawStandbyTxHandler = (
+        transactionHash: string,
+        feeCurrencyId?: string
+      ): BaseStandbyTransaction => ({
         context: newTransactionContext(TAG, 'Earn/Withdraw'),
         networkId,
         type: TokenTransactionTypeV2.EarnWithdraw,
-        inAmount: {
-          value: amount ?? pool.balance,
-          tokenId,
-        },
-        outAmount: {
-          value: amount ?? pool.balance,
-          tokenId: pool.dataProps.withdrawTokenId,
-        },
+        inAmount: { value: amount ?? pool.balance, tokenId },
+        outAmount: { value: amount ?? pool.balance, tokenId: pool.dataProps.withdrawTokenId },
         transactionHash,
         feeCurrencyId,
         providerId: pool.appId,
-      }
+      })
+      transactionHandlers.push(createWithdrawStandbyTxHandler)
     }
 
-    createWithdrawStandbyTxHandlers.push(createWithdrawStandbyTxHandler)
-
     rewardsTokens.forEach(({ balance, tokenId }, index) => {
       const createClaimRewardStandbyTx = (
         transactionHash: string,
         feeCurrencyId?: string
-      ): BaseStandbyTransaction => {
-        return {
-          context: newTransactionContext(TAG, `Earn/ClaimReward-${index + 1}`),
-          networkId,
-          amount: {
-            value: balance,
-            tokenId,
-          },
-          type: TokenTransactionTypeV2.EarnClaimReward,
-          transactionHash,
-          feeCurrencyId,
-          providerId: pool.appId,
-        }
+      ): BaseStandbyTransaction => ({
+        context: newTransactionContext(TAG, `Earn/ClaimReward-${index + 1}`),
+        networkId,
+        amount: { value: balance, tokenId },
+        type: TokenTransactionTypeV2.EarnClaimReward,
+        transactionHash,
+        feeCurrencyId,
+        providerId: pool.appId,
+      })
+
+      if (mode === 'claim-rewards' || !pool.dataProps.withdrawalIncludesClaim) {
+        transactionHandlers.push(createClaimRewardStandbyTx)
       }
-      createWithdrawStandbyTxHandlers.push(createClaimRewardStandbyTx)
     })
 
-    AppAnalytics.track(EarnEvents.earn_withdraw_submit_start, commonAnalyticsProps)
+    const eventStart =
+      mode === 'claim-rewards'
+        ? EarnEvents.earn_claim_submit_start
+        : EarnEvents.earn_withdraw_submit_start
+    AppAnalytics.track(eventStart, commonAnalyticsProps)
 
     const txHashes = yield* call(
       sendPreparedTransactions,
       serializablePreparedTransactions,
       networkId,
-      createWithdrawStandbyTxHandlers,
+      transactionHandlers,
       isGasSubsidizedForNetwork(networkId)
     )
 
     Logger.debug(
-      `${TAG}/withdraw`,
-      'Successfully sent withdraw transaction(s) to the network',
+      `${TAG}/${mode}`,
```

Comment:
nit: 
```suggestion
      `${TAG}/withdrawSubmitSaga/${mode}`,
```

Reply:
Added mode to all the logger debugs in 5086ff7bd276722149df3833f03d0286bba3254b!